### PR TITLE
Some docs cleanup

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- bumps the frontend architecture to reflect the changes from #165
- some things missing from #177, like how to build the docs

## Code changes

- travis
  - `3.8` (not `-dev`)
  - ~~add `3.9-dev`~~
    - > still getting the odd `ModuleNotFoundError: No module named 'pip'`

## User-facing changes

Some more progress towards getting the docs usable, fixing links, etc.

## Backwards-incompatible changes

N/A

## Chores

- [x] linted
- [ ] tested
- [x] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by bollwyvl** [comment]
Actually got the link checker wired up and passing locally... once this works in CI, that's probably a good place to stop!

**Comment by bollwyvl** [comment]
LGTM

File: docs/EXTENDING.md [hunk 1]
```diff
@@ -13,7 +13,7 @@
 Language Server Specs can be [configured](./LANGUAGESERVERS.md) by Jupyter users,
 or distributed by third parties as python or JSON files. Since we'd like to see
 as many Language Servers work out of the box as possible, consider
-[contributing a spec](../CONTRIBUTING.md#specs), if it works well for you!
+[contributing a spec](../CONTRIBUTING.ipynb#specs), if it works well for you!
```
Comment (krassowski) [inline docs/EXTENDING.md]: Confused by this...

File: docs/LANGUAGESERVERS.md [hunk 2]
```diff
@@ -42,12 +42,12 @@ tested to work with `jupyter-lsp`.
 Don't see an implementation for the language server you need? You can
 [bring your own language server](#adding-custom-language-servers).
 
-> Please consider [contributing your language server spec](../../CONTRIBUTING.md#spec)
+> Please consider [contributing your language server spec](../Contributing.md)
```
Comment (krassowski) [inline docs/LANGUAGESERVERS.md]: this

File: docs/LANGUAGESERVERS.md [hunk 3]
```diff
@@ -104,32 +106,32 @@ c.LanguageServerManager.language_servers = {
 }
 ```
 
-### Python `entry_points`
+#### Python `entry_points`
 
 `pip`-installable packages in the same environment as the Jupyter `notebook` server
 can be automatically detected as providing a language server spec. These are a
-little more involved: see [CONTRIBUTING](../../CONTRIBUTING.md).
+little more involved: see [CONTRIBUTING](Contributing.md).
```
Comment (krassowski) [inline docs/LANGUAGESERVERS.md]: and this.

File: docs/conf.py [hunk 4]
```diff
@@ -88,10 +88,10 @@
     "gh": (f"{github_url}/%s", "GitHub: "),
 }
 
-html_show_sourcelink = True  # True will show link to source
+html_show_sourcelink = True
 
 html_context = {
-    "display_github": True,
+    "display_github": False,
```
Comment (krassowski) [inline docs/conf.py]: IMO worth turning on (not necessarily in this PR)

**Comment by krassowski** [comment]
Looks good in general, just want to get a confirmation about the paths to Contributing. By my intuition, at least one of them is must be wrong (but I can be wrong too!).

**Comment by bollwyvl** [comment]
All of the above are link-checker-driven-development artifacts, whoops:
- i think i can fix all the internal links: (nb)sphinx is strangely passive aggressive about file extensions, in that sometimes it doesn't even care which you use, and sometimes gets very angry
  - might end up rewriting more markdown getting passed through to notebooks, or see if i can get rid of the wrappers altogether (preferable)
- the github link creates links on generated pages that don't exist, like `search` and `genindex`
  - i'll look into not showing those on those specific pages

File: docs/EXTENDING.md (hunk 1 above)
Comment (bollwyvl) [reply docs/EXTENDING.md]: fixed

File: docs/LANGUAGESERVERS.md (hunk 2 above)
Comment (bollwyvl) [reply docs/LANGUAGESERVERS.md]: fixed

File: docs/LANGUAGESERVERS.md (hunk 3 above)
Comment (bollwyvl) [reply docs/LANGUAGESERVERS.md]: fixed

File: docs/conf.py (hunk 4 above)
Comment (bollwyvl) [reply docs/conf.py]: turned on

**Comment by bollwyvl** [comment]
Well, that ended up being a bit more work than i was hoping, but it's feeling better:
- fixed the the `search` and `genindex` github links 
  - ...rather brutally by forking the template (ignored by `prettier`)
- moved all the `docs` content into notebooks
- generate the language server tables directly from a hot manager instance :tada: 
  - requires doing the full yarn and python install, but we'll KNOW the docs are right (if they build)

I guess at this point with as much content in there as there is, it may well be worth it to grab some of the stuff from `prenotebook` as part of `lint.py`:
- `black`/`isort` the python code cells
- `prettier` the markdown
- strip the outputs/execution numbers

File: docs/_templates/breadcrumbs.html [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    {% block breadcrumbs %}
+      <li><a href="{{ pathto(master_doc) }}" class="icon icon-home"></a> &raquo;</li>
+        {% for doc in parents %}
+          <li><a href="{{ doc.link|e }}">{{ doc.title }}</a> &raquo;</li>
+        {% endfor %}
+      <li>{{ title }}</li>
+    {% endblock %}
+    {% block breadcrumbs_aside %}
+      <li class="wy-breadcrumbs-aside">
+        {% if hasdoc(pagename) and pagename not in ["search", "genindex"] %}
```
Comment (bollwyvl) [inline docs/_templates/breadcrumbs.html]: all for this line...

File: README.md [hunk 6]
```diff
@@ -4,11 +4,11 @@
 
 > _This project is still maturing, but you are welcome to check it out, leave feedback and/or a PR_
 
-Quick Links: **[Installation](#installation) | [Language Servers](./docs/LANGUAGESERVERS.md) | [Updating](#updating) | [Changelog](./CHANGELOG.md) | [Roadmap](./docs/ROADMAP.md) | [Contributing](./CONTRIBUTING.md) | [Extending](./docs/EXTENDING.md)**
+Quick Links: **[Installation](./Installation.ipynb) | [Configuring](./docs/Configuring.ipynb) | [Updating](#updating) | [Changelog](./CHANGELOG.md) | [Roadmap](./docs/Roadmap.ipynb) | [Contributing](./CONTRIBUTING.md) | [Extending](./docs/Extending.ipynb)**
```
Comment (krassowski) [inline README.md]: `./docs/Installation.ipynb` ? Actually, I would probably prefer to have this go to the short version of the installation within README (we can shorten it) and then to have a link to the notebook from here.

In the quick links we maybe could have something like: `Installation: [Basic](#installation) | [Full](notebook)` (but I am not full convinced)

**Comment by bollwyvl** [comment]
Thanks for the review, looking into it now...

File: README.md (hunk 6 above)
Comment (bollwyvl) [reply README.md]: agreed, README.md needs to pretty much be standalone for the 80% case. Hopefully fixed now.

**Comment by bollwyvl** [comment]
Probably best if this lands prior to #199 so that i can fix up changelog and extending while it's still fresh in my mind... so many moving pieces.

**Comment by krassowski** [comment]
We will need to do something about these Linux/Docker failures...

File: README.md [hunk 7]
```diff
@@ -78,26 +78,60 @@ Either:
...
    pip install --pre jupyter-lsp
    ```
 
-2. install the frontend extension:
+1. install `nodejs`
+
+   ```bash
+   conda install -c conda-forge nodejs
+   # or one of the following, as an administrator
+   choco install nodejs            # Windows with Chocolatey
```
Comment (krassowski) [inline README.md]: if it is all in one block, I am 100% sure some users will just copy-paste it all. I would move these examples to the docs.

File: README.md [hunk 8]
```diff
@@ -78,26 +78,60 @@ Either:
...
+
+For the current stable version, the following steps are recommended.
+Use of a python `virtualenv` or a conda env is also recommended.
+
+1. install python 3
+
+   ```bash
+   conda install -c conda-forge python=3
+   # or
+   conda install -c conda-forge jupyterlab=1.2
```
Comment (krassowski) [inline README.md]: ?

**Pull request merged**